portfolio/src/main/webapp/blogScript.js

```diff
@@ -1,27 +1,52 @@
 window.addEventListener('DOMContentLoaded', (event) => {
+    getURL();
+    getPost();
     getComments();
 });
 
-function getComments() {
-    fetch('/data').then(response => response.json()).then((comments) => {  
-      const commentListElement = document.getElementById('comment-list');
-      for (comment of comments) {
-        commentListElement.appendChild(createCommentElement(comment));
+const tagIndx = parent.document.URL.substring(parent.document.URL.indexOf('tag='), parent.document.URL.length);
+const tagStr = tagIndx.substring(4);
+
+function getURL() {
+    const urlElement = document.getElementById('current-url');
+    urlElement.value = '/blog.html?tag=' + tagStr;
+}
+
+function getPost() {
+  fetch('/post?tag=' + tagStr).then(response => response.json()).then((posts) => {  
+    const postElement = document.getElementById('post-container');
+    const titleElement = document.getElementById('title-container');
+    postElement.innerHTML = '';
+    titleElement.innerHTML = '';
+    for (post of posts) {
+      if (post.tag === tagStr) {
```

Comment:
Since you've fixed the filtering on the server side, you don't need to do this client-side filtering anymore, right?